src/cascadia/TerminalSettingsEditor/ProfileViewModel.cpp

```diff
@@ -374,6 +391,150 @@ namespace winrt::Microsoft::Terminal::Settings::Editor::implementation
+            for (const auto sound : inheritedSounds)
+            {
+                newSounds.Append(sound);
+            }
+            _profile.BellSound(newSounds);
+        }
+    }
+
+    void ProfileViewModel::_MarkDuplicateBellSoundDirectories()
+    {
+        for (uint32_t i = 0; i < _CurrentBellSounds.Size(); i++)
+        {
+            auto soundA = _CurrentBellSounds.GetAt(i);
+            for (uint32_t j = i + 1; j < _CurrentBellSounds.Size(); j++)
+            {
+                auto soundB = _CurrentBellSounds.GetAt(j);
+                if (soundA.DisplayPath() == soundB.DisplayPath())
+                {
+                    get_self<BellSoundViewModel>(_CurrentBellSounds.GetAt(i))->ShowDirectory(true);
+                    get_self<BellSoundViewModel>(_CurrentBellSounds.GetAt(j))->ShowDirectory(true);
+                }
+            }
+        }
+    }
+
+    hstring BellSoundViewModel::DisplayPath() const
+    {
+        if (FileExists())
+        {
+            // filename
+            std::filesystem::path filePath{ std::wstring_view{ _Path } };
+            return hstring{ filePath.filename().wstring() };
+        }
+        return _Path;
+    }
+
+    hstring BellSoundViewModel::SubText() const
+    {
+        if (FileExists())
+        {
+            // Directory
+            std::filesystem::path filePath{ std::wstring_view{ _Path } };
+            return hstring{ filePath.parent_path().wstring() };
+        }
+        return RS_(L"Profile_BellSoundNotFound");
+    }
+
+    bool BellSoundViewModel::FileExists() const
+    {
+        return std::filesystem::exists(std::wstring_view{ _Path });
+    }
+
+    hstring ProfileViewModel::BellSoundPreview()
+    {
+        const auto& currentSound = BellSound();
+        if (!currentSound || currentSound.Size() == 0)
+        {
+            return RS_(L"Profile_BellSoundPreviewDefault");
+        }
+        else if (currentSound.Size() == 1)
+        {
+            std::filesystem::path filePath{ std::wstring_view{ currentSound.GetAt(0) } };
+            return hstring{ filePath.filename().wstring() };
+        }
+        return RS_(L"Profile_BellSoundPreviewMultiple");
+    }
+
+    void ProfileViewModel::_BellSoundVMPropertyChanged(const IInspectable& sender, const PropertyChangedEventArgs& args)
+    {
+        if (args.PropertyName() == L"Path")
+        {
+            auto senderVM = sender.as<Editor::BellSoundViewModel>();
+
+            // propagate changes to model
+            uint32_t index;
+            if (_CurrentBellSounds.IndexOf(senderVM, index))
+            {
+                // if current layer is inheriting,
+                // we should copy the bell sound then apply changes
+                _PrepareModelForBellSoundModification();
+
+                _profile.BellSound().SetAt(index, senderVM.Path());
+                _NotifyChanges(L"CurrentBellSounds");
+            }
+        }

```

Comment:
This isn't necessary anymore right? Since we are no longer allowing changing an entry in-line

Reply:
removed and validated we didn't need it anymore. Thanks! Good catch!